staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -652,17 +666,21 @@ func (c *Cacher) dispatchEvents() {
 func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 	triggerValues, supported := c.triggerValues(event)
 
+	// Start dispatching phase - choose watchers potentially interested
+	// in that event.
 	c.Lock()
-	defer c.Unlock()
+	c.dispatching = true
+	c.watchersBuffer = c.watchersBuffer[:0]
```

Comment:
With `[:0]` go is not allowed to GC the elements of the slice, because it's required to preserve them if the slice is later reextended. Only if subsequent appends replace elements in the underlying array can they be GCd (https://play.golang.org/p/hnrAVAr0mYc). Is that safe here?

Reply:
Thanks for explaining!